src/ifcparse/IfcParse.cpp

```diff
@@ -2260,8 +2267,15 @@ IfcUtil::IfcBaseClass* IfcFile::instance_by_guid(const std::string& guid) {
 
 // FIXME: Test destructor to delete entity and arg allocations
 IfcFile::~IfcFile() {
-	for( entity_by_id_t::const_iterator it = byid.begin(); it != byid.end(); ++ it ) {
-		delete it->second;
+	std::set<IfcUtil::IfcBaseClass*> entities_to_delete;
+	for (const auto& pair : byid) {
+		entities_to_delete.insert(pair.second);
+	}
+	for (const auto& pair : entity_file_map) {
```

Comment:
I don't think this is necessary. I can't imagine a scenario where entity_file_map contains data that is not already contained in byid. Did you encounter such cases?

Reply:
For instance, the IfcPlaneAngleMeasure instance created in `IfcHierarchyHelper::addProject()` is a type_declaration and not an entity. It will be in entity_file_map but not in byid. (from IfcAdvancedHouse sample)